teachertool/src/components/PrintButton.tsx

```diff
@@ -2,16 +2,21 @@ import * as React from "react";
 import { useRef } from "react";
 import { useReactToPrint } from "react-to-print";
 import { Toolbar } from "./Toolbar";
+import { stateAndDispatch } from "../state";
+import { showToast } from "../state/actions";
+import { makeToast } from "../utils";
 
 interface PrintButtonProps {
     printRef: React.RefObject<HTMLDivElement>;
-    onHandlePrint: () => void;
+    onHandlePrint?: () => void;
 }
 
 export const PrintButton: React.FC<PrintButtonProps> = ({ printRef, onHandlePrint }) => {
+    const { state: teacherTool } = stateAndDispatch();
     const handlePrint = useReactToPrint({
         content: () => printRef.current,
-        onAfterPrint: onHandlePrint,
+        onPrintError: () => showToast(makeToast("error", lf("Unable to print evaluation results"), 2000)),
+        documentTitle: `MakeCode Project Evaluation Results - ${teacherTool.projectMetadata?.name}`,
```

Comment:
I'm guessing we'll want to sanitize the `teacherTool.projectMetadata.name` here in case it contains invalid file name characters. We already have a `sanitizeFileName` helper function for this in `pxt.Util`. We use it when exporting rubrics too: https://github.com/microsoft/pxt/blob/thsparks/teachertool/better_ui_for_customizable_criteria/teachertool/src/services/fileSystemService.ts#L8

Reply:
Yeah, I can add this. 

Just want some quick brainstorming for this. Since `sanitizeFileName` needs a string, I need  to make sure that the project metadata exists before passing in `teacherTool.projectMetadata.name`, but it should be impossible for a user to print without project metadata. That said, if they do get to a case where they can see the print button without a project loaded, I figure we should error and say something like "cannot print results if there is no project loaded". If I do error, I figure I should move the "sanitize" logic out of the component, but I'm unsure of the best place to put it. Maybe as a transform or in utils?